### ISSUE TITLE
Error when Deploying API portal locally

### ISSUE THREAD

**Issue by bigdoods** [Haufe-Lexware/wicked.haufe.io#61]
Hi,

I have followed instructions. Everything was fine until I ran docker-compose up and I received this error:

```
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.portal-api.environment contains non unique items, please remove duplicates from ['NODE_ENV', 'DEBUG', None, None, 'PORTAL_CONFIG_KEY', 'GIT_REPO', 'GIT_BRANCH', 'GIT_CREDENTIALS', 'GIT_REVISION']
```

I tried to remove the corresponding empty lines. When I did this, I received a 503 Service Unavailable - No server is available to handle this request.

**Comment by hlgr360** [comment]
Could you attach the docker-compose file and send a trace of your last shell commands prior to invoking `docker-compose up`

**Comment by DonMartin76** [comment]
I can reproduce the problem; there is a regression in there regarding the templating of the `docker-compose.yml` file, sorry for that, will fix in the next version.

What's more interesting is that I can also reproduce the 503 regarding the `https://portal.local`, which seems to be a change in docker 1.13 (17.03) that does not fit the HAproxy version. I will investigate and come back.

Can you please attach your local machine's configuration:

* Operating system and version
* `docker` version, 
* `docker-compose` version

I am guessing you're on macOS?

**Comment by DonMartin76** [comment]
Short headsup: It's definitely something with the HAproxy. It doesn't map the portal out; funnily enough, the API Gateway is routed, which can be tested with

```
$ curl --insecure https://api.portal.local/ping-portal
```

This answers correctly, and it shows that both the gateway and the portal are running correctly _internally_, and it's "just" the frontend load balancer which seems to be misconfigured. Will continue investigating, this is weird, as it has worked exactly like this until (probably) very recently.

**Comment by DonMartin76** [comment]
*Blushing*

I found the problem: In your `docker-compose.yml`, check the `portal:` section. In this section, you will find two `environment:` sections, one with `VIRTUAL_HOST` and others, and one with only `- DEBUG`. Move the `- DEBUG` into the first one and delete the second `environment:` section.

I assumed `docker-compose` would merge the two `environment` sections, or at least throw an error if it finds multiple ones. This is apparently not the case, and this one slipped through the tests.

Will fix this for the next version. Sorry for the inconvenience.

**Comment by bigdoods** [comment]
Hi,

This worked but I also had to remove the two blank fields in the environment section of portal-api in the docker-compose.

Thanks for quick resolution.

John

**Comment by DonMartin76** [comment]
Yes, thanks, I fixed that as well with the commit above. Will schedule a release for when I'm back from KubeCon. Sorry.

Martin

**Comment by DonMartin76** [comment]
Fixed and released in 0.11.5.
